lib/kafka/consumer.rb

```diff
@@ -244,6 +244,7 @@ def each_message(min_bytes: 1, max_wait_time: 1, automatically_mark_as_processed
     # @return [nil]
     def each_batch(min_bytes: 1, max_wait_time: 1, automatically_mark_as_processed: true, exit_if: nil)
       consumer_loop do
+        @running = false if exit_if && exit_if.call
```

Comment:
Shouldn't you `return` or `break` here? otherwise it's still gonna execute the next `fetch_batches` even though we already know that we want to exit.